mypy/checker.py

```diff
@@ -4534,10 +4535,11 @@ def refine_identity_comparison_expression(self,
         if singleton_index == -1:
             singleton_index = possible_target_indices[-1]
 
-        enum_name = None
+        sum_type_name = None
         target = get_proper_type(target)
-        if isinstance(target, LiteralType) and target.is_enum_literal():
-            enum_name = target.fallback.type.fullname
+        if (isinstance(target, LiteralType) and
+                (target.is_enum_literal() or target.fallback.type.fullname == "builtins.bool")):
```

Comment:
is there a better way to check if the type is a bool than looking for this string?

Reply:
`isinstance(target.value, bool)`?